code/game/turfs/open/water.dm

```diff
@@ -121,18 +119,18 @@
 	AddElement(/datum/element/immerse, icon, icon_state, "immerse", immerse_overlay_color, alpha = immerse_overlay_alpha)
 	immerse_added = TRUE
 	icon_state = "pool_[rand(1, 4)]"
-	particle_effect = new(src, /particles/hotspring_steam, 4)
-	//render the steam over mobs and objects on the game plane
-	particle_effect.vis_flags &= ~VIS_INHERIT_PLANE
-	//And be unaffected by ambient occlusions, which would render the steam grey
-	particle_effect.plane = MUTATE_PLANE(MASSIVE_OBJ_PLANE, src)
+	var/obj/effect/abstract/shared_particle_holder/holder = add_shared_particles(/particles/hotspring_steam, "hot_springs_[GET_TURF_PLANE_OFFSET(src)]", pool_size = 4)
+	// Render the steam over mobs and objects on the game plane
+	holder.vis_flags &= ~VIS_INHERIT_PLANE
```

Comment:
this seems... wrong to do this way, why do we have to do this like this?

Reply:
that'd be because by default they need to inherit parent's plane, but then they'd render below mobs

Reply:
I so wish we had IMPLIED_TYPE as a var instead of a fucked macro man

Reply:
saaame